data/StoreSelectionModel.js

```diff
@@ -35,29 +35,26 @@ export class StoreSelectionModel extends HoistModel {
         this.addReaction(this.cullSelectionReaction());
     }
 
-    /** @return {?Record} - single selected record, or null if multiple or no records selected. */
-    @computed
-    get singleRecord() {
-        const ids = this.ids;
-        return ids.length === 1 ? this.store.getById(ids[0]) : null;
-    }
-
     /** @return {Record[]} - currently selected Records. */
-    @computed
+    @computed({equals: comparer.shallow})
     get records() {
         return compact(this.ids.map(it => this.store.getById(it)));
```

Comment:
Checking to ensure I understand - will this re-compute if the recordset / records change, even if `this.ids` does not?

I am thinking it will because `store.getById()` is accessing the recordSet and therefore mobx knows this depends on it. And then the `comparer.shallow` will detect if any of the selected records themselves have actually changed (given that they might have been left unchanged and would therefore be reused across store updates).

Reply:
yep